### PR TITLE
Add registerInterfaceKernel in Factory.h

### LINKED ISSUE #6434: Add registerInterfaceKernel
To be consistent with other base classes, add the registerInterfaceKernel definition in Factory.h

### PR DESCRIPTION
**Pull request by lindsayad**
Makes the registration of interface kernels consistent with other base classes

Closes #6434

### CODE REVIEW

File: framework/include/base/Factory.h [hunk 1]
```diff
@@ -47,6 +47,7 @@
 #define registerDamper(name)                        registerObject(name)
 #define registerDiracKernel(name)                   registerObject(name)
 #define registerDGKernel(name)                      registerObject(name)
+#define registerInterfaceKernel(name)                      registerObject(name)
```
Comment (jwpeterson) [inline framework/include/base/Factory.h]: This triggered my OCD.

Comment (lindsayad) [inline framework/include/base/Factory.h]: Lol, I don't know how I missed that. Will fix

**Comment by permcody** [comment]
Thanks! This test failure isn't your fault. We have an open ticket to fix this #6429.

**Comment by permcody** [comment]
:+1:

**Pull request merged**